### PR TITLE
Migrate to ROS2 Humble with ROS Action support

### PR DESCRIPTION
**Pull request by MishkaMN**
# PR Details
## Description
This PR brings our fork up to date with the ros2 repo that allows the ros1_bridge to build with all the autoware and carma messages. 
This PR also merges ROS action support from here https://github.com/smith-doug/ros1_bridge/tree/action_bridge_humble
This PR also updates all github action versions and removes `required-ros-distributions: "noetic rolling"` which is intended to install the ROS versions from distribution because ROS1 noetic is not available on 22.04 anymore. By removing it, the `setup-ros` action installs it from source. 

## Related GitHub Issue
NA

## Related Jira Key
[ARC-156](https://usdot-carma.atlassian.net/browse/ARC-156)

## Motivation and Context
Upgrade to ROS2 Humble

## How Has This Been Tested?
Built using dockerhub using carma-msgs Dockerifle in https://github.com/usdot-fhwa-stol/carma-msgs/pull/248
And tested in cdasim xil carma cloud use case 

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [X] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [X] I have added any new packages to the sonar-scanner.properties file
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING**](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

[ARC-156]: https://usdot-carma.atlassian.net/browse/ARC-156?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: CODEOWNERS [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# This file was generated by https://github.com/audrow/update-ros2-repos
+* @methylDragon @quarkytale @gbiggs
```
Comment (john-chrosniak) [inline CODEOWNERS]: It looks like github is complaining since these developers don't have write access to our repo, could we remove this?

**Comment by john-chrosniak** [comment]
It looks like CI is failing when running `sudo apt-get update` because it is looking for a ROS1 installation on 22.04. Is that due to [carma-msgs #248](https://github.com/usdot-fhwa-stol/carma-msgs/pull/248) not being merged yet?

File: CODEOWNERS (hunk 1 above)
Comment (MishkaMN) [reply CODEOWNERS:2]: yeah we can

Comment (MishkaMN) [reply CODEOWNERS]: yeah we can. removed.

**Comment by MishkaMN** [comment]
> It looks like CI is failing when running `sudo apt-get update` because it is looking for a ROS1 installation on 22.04. Is that due to [carma-msgs #248](https://github.com/usdot-fhwa-stol/carma-msgs/pull/248) not being merged yet?

Looks like the Github Action versions were outdated. I updated all github action versions and removed `required-ros-distributions: "noetic rolling"` which is intended to install the ROS versions from distribution because ROS1 noetic is not available on 22.04 anymore. By removing it, the `setup-ros` action installs it from source.

**Review by john-chrosniak — APPROVED** [review]

**Pull request merged**